plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/LazyConnectionFactory.java

```diff
@@ -39,6 +39,7 @@ public LazyConnectionFactory(@ForLazyConnectionFactory ConnectionFactory delegat
     }
 
     @Override
+    @SuppressWarnings("CheckedExceptionNotThrown") // throws in the interface declaration; keep it to prevent false sense of security
```

Comment:
i don't understand the comment message

Reply:
BTW this seems not violate the check
```
public interface AnInterface
{
    void foo()
            throws Exception;
}

public class AnImplementationThatCurrentlyDoesNotThrow
        implements AnInterface
{
    @Override
    public void foo()
            throws Exception
    {}
}
```

so why do we need suppression here?
is it a bug in error-prone?

Reply:
Turns out that this check does trigger in this case if the class is `final`, otherwise it does not. See this comment: https://github.com/google/error-prone/issues/3425#issuecomment-1319344180